internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1070,6 +1073,42 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, log logrus.Field
 		// If none platform do not attempt to create spoke BMH
 		return reconcileComplete{}
 	}
+	isBaremetalPlatform, err := isBaremetalPlatform(ctx, r.Client, cd)
```

Comment:
I think if we extract this logic to something like `skipSpokeBMH, err := isBaremetalPlatformWithoutMAPI(...)` then the overall method will be easier to follow.

Reply:
Modified as suggested! Thanks!